.github/workflows/macos.yml

```diff
@@ -0,0 +1,107 @@
+---
+name: MacOS build
+
+on:
+  push:
+    branches:
+      - "jhale/macos-homebrew-build"
+  schedule:
+    # '*' is a special character in YAML, so string must be quoted
+    - cron: "0 4 * * WED"
+  workflow_dispatch: ~
+
+jobs:
+  mac-os-build:
+    name: MacOS Homebrew install and test
+    strategy:
+      matrix:
+        os: [macos-12, macos-11]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install Homebrew dependencies
+        run: |
+          brew install boost cmake hdf5-mpi ninja open-mpi pkg-config pugixml python@3.10 xtensor # FEniCS
```

Comment:
It looks a bit weird installing Python with an 'action' and with Homebrew.

Reply:
Apparently the action 'activates' the desired Homebrew Python, rather than installing it.

To be explicit for people who want to understand how FEniCS is built on MacOS, I left in dependencies that are already installed in the runner images. I'd prefer clean images but...

I can add comments to this effect.